src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -360,7 +360,7 @@ internal BoundExpression BindValueAllowArgList(ExpressionSyntax node, Diagnostic
             out BoundExpression valueBeforeConversion)
         {
             Debug.Assert(this.InParameterDefaultValue);
-            Debug.Assert(this.ContainingMember().Kind == SymbolKind.Method || this.ContainingMember().Kind == SymbolKind.Property);
+            Debug.Assert(this.ContainingMemberOrLambda.Kind == SymbolKind.Method || this.ContainingMemberOrLambda.Kind == SymbolKind.Property);
```

Comment:
Right. Then it seems the original check (`this.ContainingMember().Kind == SymbolKind.Property`) was good, no?

---
In reply to: [98312669](https://github.com/dotnet/roslyn/pull/16789#discussion_r98312669) [](ancestors = 98312669)